patches/tModLoader/Terraria/ModLoader/BuildInfo.cs

```diff
@@ -37,23 +38,24 @@ public enum BuildPurpose
 		static BuildInfo() {
 			var parts = BuildIdentifier.Substring(BuildIdentifier.IndexOf('+')+1).Split('|');
 			tMLVersion = new Version(parts[0]);
-			if (parts.Length>=2) {
-				BranchName = parts[1];
+			ltsVersion = new Version(parts[1]);
+			if (parts.Length>=3) {
+				BranchName = parts[2];
 			}
 			else {
 				BranchName = "unknown";
 			}
-			if (parts.Length>=3) {
-				Enum.TryParse(parts[2], true, out Purpose);
-			}
 			if (parts.Length>=4) {
-				CommitSHA = parts[3];
+				Enum.TryParse(parts[3], true, out Purpose);
+			}
+			if (parts.Length>=5) {
+				CommitSHA = parts[4];
 			}
 			else {
 				CommitSHA = "unknown";
 			}
-			if (parts.Length>=5) {
-				BuildDate = DateTime.FromBinary(long.Parse(parts[4])).ToLocalTime();
+			if (parts.Length>=6) {
+				BuildDate = DateTime.FromBinary(long.Parse(parts[5])).ToLocalTime();
```

Comment:
Do we need all of these `parts.Length` checks? It looks like the number of separators is fixed now, so we could do fixed indices, or `parts[i++]` in all of them

Reply:
No we really don't. I probably should have removed them when I touched it; all the safety is in the csproj now.

Updated accordingly.